pkg/util/filepath_test.go

```diff
@@ -16,16 +17,16 @@ func TestCleanRelativePath(t *testing.T) {
 			expectedPath: ".",
 		},
 		{
-			input:        "/test/test.txt",
-			expectedPath: "test/test.txt",
+			input:        filepath.Join(string(filepath.Separator), "test", "test.txt"),
+			expectedPath: filepath.Join("test", "test.txt"),
 		},
 		{
-			input:        "../../test/test.txt",
-			expectedPath: "test/test.txt",
+			input:        filepath.Join("..", string(filepath.Separator), "..", string(filepath.Separator), "test", "test.txt"),
+			expectedPath: filepath.Join("test", "test.txt"),
 		},
 		{
-			input:        "./../test/test.txt",
-			expectedPath: "test/test.txt",
+			input:        filepath.Join(".", string(filepath.Separator), "..", string(filepath.Separator), "test", "test.txt"),
```

Comment:
It looks like this returns a different value rather than the old `"./../test/test.txt"`

```go
filepath.Join(".", string(filepath.Separator), "..", string(filepath.Separator), "test", "test.txt")
```

```
../test/test.txt
```

https://goplay.tools/snippet/kxUmCCGoW6_l

I think we have to change this to:

```go
"." + string(filepath.Separator) + filepath.Join("..", "test", "test1.txt")
```

```
./../test/test1.txt
```

Reply:
Confused by this 🤔 Have updated the test case though - is that not what we're looking for?

Reply:
The old test case had this `input` string: `"./../test/test.txt"`

`filepath.Join` also applies `filepath.Clean`:

https://pkg.go.dev/path/filepath#Clean

From the docs, you can see `Clean` removes all elements that are `.`

Meaning that

```go
filepath.Join(".", string(filepath.Separator), "..", string(filepath.Separator), "test", "test.txt")
```

returns

```
../test/test.txt
```

instead of

```
./../test/test.txt
```

so we technically have a different test case compared to the code that's currently in main.

Does that make sense?

Reply:
Ah yes forgot about our good friend `Clean`. Good spot mate!